### PR TITLE
fix: http response empty body

### PR DESCRIPTION
**Pull request by TheEdward162**
* NodeJS host was failing when fetch request returned an empty body
* Python handles this okay
* Normalized so that the core always gets a stream, but when the body is empty the stream reads 0 bytes

### CODE REVIEW

File: scripts/build-a-core/template.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def dockerfile_template(
 
 WORKDIR /opt/wasi-sdk
 ARG WASI_SDK_REF={wasi_sdk_ref}
-RUN git clone https://github.com/WebAssembly/wasi-sdk.git . && git checkout $WASI_SDK_REF && git submodule update --init --depth 10
+RUN git clone https://github.com/WebAssembly/wasi-sdk.git . && git checkout $WASI_SDK_REF && git submodule update --init --depth 50
```
Comment (freaz) [inline scripts/build-a-core/template.py:37]: Why we need it increased?

**Review by freaz — APPROVED** [review]
Thank you

File: scripts/build-a-core/template.py (hunk 1 above)
Comment (TheEdward162) [reply scripts/build-a-core/template.py:37]: Submodule dependencies are (apparently) checked out by first cloning the repository and then checking out the commit the submodule points at.

One of the second-level submodule dependencies added more commits into its default branch so the top 10 did not include the commit that was supposed to get checked out, so we need to clone more commits to get the latest commit in there. Worst case scenario we need to clone the whole repo.

Comment (freaz) [reply scripts/build-a-core/template.py:37]: I see, thanks for clarification

**Pull request merged**